.github/workflows/docker.yml

```diff
@@ -12,9 +12,50 @@ on:
     - 'Cargo.lock'
     - 'Dockerfile'
 
+env:
+  DOCKER_UNOPTIMIZED: "1"
+
 jobs:
   docker:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v1
-    - run: make docker
+
+    - name: Docker (Fork)
+      if: github.event.pull_request.head.repo.fork
+      run: make docker
+
+    # Create a build image on a Linkerd build host.
+    - name: Setup (Origin)
```

Comment:
nit: consider moving `Docker (Fork)` step next to `Docker (Origin)`. no functional difference just cognitively that's where it sits in step ordering.